connector/connect/common/src/main/scala/org/apache/spark/sql/connect/common/DataTypeProtoConverter.scala

```diff
@@ -177,7 +178,11 @@ object DataTypeProtoConverter {
       case s: StringType =>
         proto.DataType
           .newBuilder()
-          .setString(proto.DataType.String.newBuilder().setCollationId(s.collationId).build())
+          .setString(
+            proto.DataType.String
+              .newBuilder()
+              .setCollation(CollationFactory.fetchCollation(s.collationId).collationName)
```

Comment:
does it mean that the Spark Connect client already knows the collation id, but we turn it back to name when sending proto messages to the server?

Reply:
It's true that we use collation name in protobuf messages, this is also part of the changes (before we sent collation id).

We decided to go with this approach in order to remove collation id logic from PySpark side because if collation id is part of the protobuf message, we would need to have collation name parser logic in Python as well besides Scala (`CollationFactory`) which would be unnecessary overhead.

Reply:
I'm not sure how is this possible. `StringType` can be accessed in the client side (Scala client) and it needs to parse the collation name to ID.

Reply:
or is it specific to the python client?

Reply:
This file is in `common` directory, so it's used by both client and server, right? If I got it correctly, the utility function `toConnectProtoType` would be used by server to serialize `StringType` with collation and send it to the PySpark client.